torch/quantization/custom_module_class_mappings.py

```diff
@@ -0,0 +1,75 @@
+OBSERVED_CUSTOM_MODULE_CLASS_MAPPINGS = dict()
+
+def register_observed_custom_module_mapping(custom_module_class, observed_custom_module_class):
+    """ Register a mapping from `custom_module_class` to
+    `observed_custom_module_class`
+    `observed_custom_module_class` will have a `from_float` classmethod,
+    which will return an observed custom module instance given
+    a float custom module instance.
+    This will be used in prepare step of post training static quantization or
+    quantization aware training
+    """
+    assert hasattr(observed_custom_module_class, 'from_float'), 'from_float must be' + \
+        ' defined in observed custom module class'
+    OBSERVED_CUSTOM_MODULE_CLASS_MAPPINGS[custom_module_class] = \
+        observed_custom_module_class
+
+def get_observed_custom_module_class(custom_module_class):
+    """ Get the corresponding observed module class for a given
+    custom module.
+    """
+    observed_custom_module_class = \
+        OBSERVED_CUSTOM_MODULE_CLASS_MAPPINGS.get(custom_module_class, None)
+    assert observed_custom_module_class is not None, \
+        'Custom module class {}'.format(custom_module_class) + \
+        ' does not have a corresponding observed module class'
+    return observed_custom_module_class
+
+QUANTIZED_CUSTOM_MODULE_CLASS_MAPPINGS = dict()
+
+def register_quantized_custom_module_mapping(custom_module_class, quantized_custom_module_class):
+    """ Register a mapping from `custom_module_class` to `quantized_custom_module_class`
+    A quantized custom module class should accept quantized input and
+    return quantized output. (we can relax this condition in the
+    future if there is a need)
+    `quantized_custom_module_class` will have a `from_observed` classmethod,
+    which will return an quantized custom module instance given
+    a observed custom module instance.
+    This will be used in prepare step of post training static quantization or
+    quantization aware training
+    """
+    assert hasattr(quantized_custom_module_class, 'from_observed'), 'from_observed' + \
+        ' must be defined in quantized custom module class'
+    QUANTIZED_CUSTOM_MODULE_CLASS_MAPPINGS[custom_module_class] = \
+        quantized_custom_module_class
+
+def get_quantized_custom_module_class(custom_module_class):
```

Comment:
seems like `custom` could mean either `custom float observed module` or `custom quantized module` in this context, and it might be confusing to the user which one is used where.  Would it be possible to make the names unambiguous?